### PR TITLE
Referrables in ParameterDescriptors

### PR DESCRIPTION
**Pull request by brett-groshong**
### Purpose of this PR
This change adds referrables and referrable option families to the descriptor language.
See: `REF.cs`

For an example, see `TestUIReferrablesNode` in `TestUINodes.cs`

---
### Testing status
This change adds syntax options to the descriptor language.
All functionality should be the same as on `sg2/staging-master` branch.

---
### Comments to reviewers
- This does not implement the linking behavior of referrables, only the ability to specify them in `NodeDescriptors`.
- The instances in `REF` are not currently assigned tags. The tag fields are empty strings.

### CODE REVIEW

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/REF.cs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            {
+                ("Absolute World Space", AbsoluteWorldSpace_Bitangent),
+                ("World Space", WorldSpace_Bitangent),
+                ("Object Space", ObjectSpace_Bitangent),
+                ("View Space", ViewSpace_Bitangent),
+                ("Tangent Space", TangentSpace_Bitangent)
+            };
+            public static readonly List<(string, object)> ViewDirections = new()
+            {
+                ("Absolute World Space", AbsoluteWorldSpace_ViewDirection),
```
Comment (bencloward) [inline com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/REF.cs:43]: ViewDirection doesn't currently have an Absolute World option.

**Review by bencloward — APPROVED** [review]

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/REF.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            {
+                ("Absolute World Space", AbsoluteWorldSpace),
+                ("World Space", WorldSpace),
+                ("Object Space", ObjectSpace),
+                ("View Space", ViewSpace),
+                ("Tangent Space", TangentSpace)
+            };
+            public static readonly List<(string, object)> Normals = new()
+            {
+                ("Absolute World Space", AbsoluteWorldSpace_Normal),
```
[the remark this answers is not in the document]
Comment (esmelusina) [reply com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/REF.cs:19]: Absolute world is mainly just an HDRP thing, right?

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/VALUE.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace UnityEditor.ShaderGraph.Defs
+{
+    internal class VALUE
+    {
+        
+    }
+}
```
Comment (esmelusina) [inline com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/VALUE.cs:7]: Is this intended for use?

Comment (brett-groshong) [reply com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/VALUE.cs:7]: It was needed for completing the idea of having default values that aren't REFs.

Comment (brett-groshong) [reply com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/VALUE.cs]: Removed this file. We will address this need when it comes up.

**Review by esmelusina — APPROVED** [review]

**Pull request merged**